### PR TITLE
UserAdd changes for Staff/StaffReadOnly

### PR DESCRIPTION
**Pull request by sam9111**
- Staff/StaffReadOnly users can now select facilities from a dropdown while they are creating new users.
- If  Staff/StaffReadOnly are creating  Staff/StaffReadOnly new users, at least one of their facilities will have to be selected.
- Staff can only create Volunteer, Pharmacist, Staff, StaffReadOnly users while StaffReadOnly can create only a StaffReadOnly user.

### CODE REVIEW

**Comment by samholics** [comment]
tested, verified

File: src/Components/Common/HelperInputFields.tsx [hunk 1]
```diff
@@ -294,37 +295,42 @@ export const SelectField = (props: DefaultSelectInputProps) => {
...
+                        </div>
+                    )}
+                >
+                    {options.map((opt: any) => {
+                        const selected = value as Array<any>;
+                        return (<MenuItem key={opt.id} value={opt[optKey]}>
+                            <Checkbox checked={selected.indexOf(opt[optKey]) > -1} />
+                            <ListItemText primary={opt[optVal]} />
+                        </MenuItem>
+                        )
```
Comment (smartclash) [inline src/Components/Common/HelperInputFields.tsx]:
```suggestion
                        return (
                            <MenuItem key={opt.id} value={opt[optKey]}>
                                <Checkbox checked={selected.indexOf(opt[optKey]) > -1} />
                                <ListItemText primary={opt[optVal]} />
                            </MenuItem>
                        )
```

File: src/Components/Users/UserAdd.tsx [hunk 2]
```diff
@@ -247,21 +248,23 @@ export const UserAdd = (props: UserProps) => {
...
+
   const validateForm = () => {
     let errors = { ...initError };
     let invalidForm = false;
     Object.keys(state.form).forEach(field => {
       switch (field) {
         case "facilities":
-          if (userType === "Staff" && state.form["user_type"] === "Staff") {
+
+          if (state.form[field].length === 0 && (userType === "Staff" || userType === "StaffReadOnly") && (state.form["user_type"] === "Staff" || state.form["user_type"] === "StaffReadOnly")) {
```
Comment (smartclash) [inline src/Components/Users/UserAdd.tsx:265]: Can we refactor this long `if` conditions?

**Review by smartclash — CHANGES_REQUESTED** [review]

File: src/Components/Users/UserAdd.tsx (hunk 2 above)
Comment (sam9111) [reply src/Components/Users/UserAdd.tsx:265]: For now, this is the only use case that invalidates the form. How else can we change it? @smartclash

Comment (smartclash) [reply src/Components/Users/UserAdd.tsx:265]: Probably move that condition to a function and then add roles into an array while conditionally looping through the array? 

Can write up a suggestion tomorrow when I get back to my PC

Comment (sam9111) [reply src/Components/Users/UserAdd.tsx:265]: @smartclash any suggestions?

**Pull request merged**